shift/src/main/java/org/coursera/android/shift/ShiftPersistenceManager.java

```diff
@@ -14,51 +14,31 @@
 package org.coursera.android.shift;
 
 import android.content.Context;
+import android.content.SharedPreferences;
 import android.util.Log;
 
-import com.snappydb.DB;
-import com.snappydb.SnappyDB;
-import com.snappydb.SnappydbException;
+import com.google.gson.Gson;
 
+import java.util.Map;
 import java.util.Set;
 
-/**
- * Simple Persistence storage using https://github.com/nhachicha/SnappyDB.
- * Chose to use SnappyDB due to its simple NoSQL approach.
- * Allows us to store primitives easily like SharedPreferences, but also gives
- * flexibility to store more complicated objects in the future (Eg. Arrays with a selected index).
- */
-
 @SuppressWarnings("SynchronizeOnNonFinalField")
 class ShiftPersistenceManager {
 
     private static final String TAG = ShiftPersistenceManager.class.getCanonicalName();
 
-    private static String TABLE_NAME = "shift.persistence.";
+    private static final String SHARED_PREF_KEY = "ShiftPrefsFile";
 
     // Invalidate once per launch
     private boolean shouldInvalidate = true;
 
-    private DB db;
-
-    /*
-        SnappydbException is thrown if we are unable to create the file (out of memory)
-        or the existing DB file somehow becomes inaccessible:
-        https://github.com/nhachicha/SnappyDB/blob/4e0c0805bf5c9c157042f61d9c11842c784cf91c/library/src/main/java/com/snappydb/SnappyDB.java
-        On out of memory client's app wouldn't be able to run either and the existing DB file
-        should never have it's permissions changed by another application unless the client chooses to change it themselves.
+    private SharedPreferences sharedPreferences;
 
-        If for some reason db is null, all GET calls will still be safe because of the ability to
-        return the default value. Setting the values will not do anything.
-     */
     public ShiftPersistenceManager(Context context) {
-        TABLE_NAME = TABLE_NAME + context.getPackageName();
-        try {
-            db = new SnappyDB.Builder(context)
-                    .name(TABLE_NAME)
-                    .build();
-        } catch (SnappydbException e) {
-            Log.e(TAG, "Failed to create SnappyDB! This should never happen unless out of memory. " + e.toString());
+        if (context == null) {
```

Comment:
I think now this is simplified to shared preferences, passing in a null context is a developer error and should result in a runtime exception. You can then remove all the null checks in the code below